src/3d/qgs3dmapscene.cpp

```diff
@@ -361,6 +363,34 @@ void addQLayerComponentsToHierarchy( Qt3DCore::QEntity *entity, const QVector<Qt
 void Qgs3DMapScene::updateScene()
 {
   QgsEventTracing::addEvent( QgsEventTracing::Instant, QStringLiteral( "3D" ), QStringLiteral( "Update Scene" ) );
+
+  const QSize size = mEngine->size();
+  const int screenSize = std::max( size.width(), size.height() );
+  const float fov = mCameraController->camera()->fieldOfView();
+  const QVector3D &cameraPosition = mCameraController->camera()->position();
+  for ( QgsVirtualPointCloudEntity *entity : std::as_const( mVirtualPointCloudEntities ) )
```

Comment:
This looks very provider specific logic to sit in a generic class. Could it be moved to a virtual "handleSceneUpdate" method in the entity?

Reply:
I introduced a Qgs3DMapSceneEntity class, common ancestor to chunked entities and virtual point cloud entities and moved the logic. It also handles computing the entities' near/far plane.
Qgs3DMapScene is quite cleaner indeed, let me know what you think!